### PR TITLE
gh-111159: Fix `doctest` output comparison for exceptions with notes

### PR DESCRIPTION
**Pull request by sobolevn**
Do we need more tests for this? What else can be added?

* Issue: gh-111159

### CODE REVIEW

File: Lib/test/test_doctest.py [hunk 1]
```diff
@@ -3212,6 +3212,65 @@ def test_run_doctestsuite_multiple_times():
     """
 
 
+def test_exception_with_note(note):
+    """
+    It was failing until https://github.com/python/cpython/issues/111159
```
Comment (iritkatriel) [inline Lib/test/test_doctest.py]: Not sure this comment is necessary.

File: Lib/doctest.py [hunk 2]
```diff
@@ -1393,7 +1393,13 @@ def __run(self, test, compileflags, out):
 
             # The example raised an exception:  check if it was expected.
             else:
-                exc_msg = traceback.format_exception_only(*exception[:2])[-1]
+                formatted_ex = traceback.format_exception_only(*exception[:2])
+                if issubclass(exception[0], SyntaxError):
+                    # SyntaxError / IndentationError is special:
+                    # we don't care about the carets / suggestions / etc
+                    exc_msg = formatted_ex[-1]
```
Comment (iritkatriel) [inline Lib/doctest.py]: Is this correct? What about the notes? I think we need a test for SyntaxError because it's different.

**Comment by iritkatriel** [comment]
Thanks for doing this. Are you up for adding Exception Groups to doctest too? https://github.com/python/cpython/issues/104150

File: Lib/doctest.py (hunk 2 above)
Comment (sobolevn) [reply Lib/doctest.py]: Indeed, it is not. SyntaxError with notes does not pass. Fixing!

**Comment by sobolevn** [comment]
> Are you up for adding Exception Groups to doctest too?

I will take a look :)

File: Lib/test/test_doctest.py [hunk 3]
```diff
@@ -3212,6 +3212,150 @@ def test_run_doctestsuite_multiple_times():
...
+    TabError: error
+    Note
+    Line
+    """
+    exc = cls("error", ("x.py", 23, None, "bad syntax"))
+    exc.add_note('Note\nLine' if multiline else 'Note')
+    raise exc
+
+
+def test_syntax_error_with_incorrect_note():
```
Comment (iritkatriel) [inline Lib/test/test_doctest.py]:
```suggestion
def test_syntax_error_with_incorrect_expected_note():
```

(because this is how the test is written).

**Review by iritkatriel — APPROVED** [review]

**Pull request merged**